src/transpose.cpp

```diff
@@ -0,0 +1,140 @@
+#include <Rcpp.h>
+using namespace Rcpp;
+
+// copy from x with number of rows, to c column in columns
+template <typename T, class RTYPE>
```

Comment:
This is not `RTYPE` but rather `CPPTYPE`?

I'm also not crazy about the name `copy`, but I don't have a better suggestion off the top of my mind.